### PR TITLE
[RDBMS] `az mysql flexible-server create/update/restore/geo-restore/replica`: Add BYOK for MySQL Flexible Server

### PR DESCRIPTION
**Pull request by alanenriqueo**
**Related commands**
`az mysql flexible-server create --key key --identity identity`
`az mysql flexible-server update --key key --identity identity`
`az mysql flexible-server restore`
`az mysql flexible-server geo-restore`
`az mysql flexible-server replica`

**Description**
This PR enables data encryption for MySQL Flexible Server with customer-managed identity and customer-managed key.

- Add arguments `--key` and `--identity` for create and update server.
- Add argument `--disable-data-encryption` for update server.
- Send key and identity properties for replica create, restore from a backup and geo-restore.
- Add arguments `--backup-key` and `--backup-identity` for all of the above, for geo-redundant servers.

**Testing Guide**
See [_helptext_mysql.py](https://github.com/Azure/azure-cli/pull/23197/files#diff-e7e0e994269ad38b069f8618e75c536cf5a7ae0fe79514277bbb332f00cee7afR96) and [test_rdbms_flexible_commands.py](https://github.com/Azure/azure-cli/pull/23197/files#diff-64267f415c845d242b32853a748c529022a3f513732312caec287632cf5a26b3R509)

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
RDBMS

File: src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_mysql.py [hunk 1]
```diff
@@ -331,7 +355,11 @@ def flexible_server_update_custom_func(cmd, client, instance,
                               standby_availability_zone=standby_availability_zone,
                               auto_grow=auto_grow,
                               replication_role=replication_role,
-                              instance=instance)
+                              instance=instance,
+                              byok_identity=byok_identity,
+                              backup_byok_identity=backup_byok_identity,
```
Comment (DaeunYim) [inline src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_mysql.py:360]: We are not allowing this for backup iteration? Or Cong said to open this parameter open for customer? (Only not exposing georestore, restore coammdns with backup parameter and leaving these params)

File: src/azure-cli/azure/cli/command_modules/rdbms/tests/latest/recordings/test_mysql_flexible_server_byok_mgmt.yaml [hunk 2]
```diff
@@ -0,0 +1,3489 @@
...
+      CommandName:
+      - keyvault create
+      Connection:
+      - keep-alive
+      ParameterSetName:
+      - -g -n --location --enable-soft-delete --enable-purge-protection
+      User-Agent:
+      - AZURECLI/2.38.0 azsdk-python-azure-mgmt-keyvault/9.3.0 Python/3.8.10 (Linux-5.10.102.1-microsoft-standard-WSL2-x86_64-with-glibc2.29)
+    method: GET
+    uri: https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/clitest.rg000001/providers/Microsoft.KeyVault/vaults/rdbmsvault000002?api-version=2021-06-01-preview
```
Comment (DaeunYim) [inline src/azure-cli/azure/cli/command_modules/rdbms/tests/latest/recordings/test_mysql_flexible_server_byok_mgmt.yaml:18]: Wanted to make sure you notice that test recordings mask the resource names like this. In some cases where you need the exact names for the resources in the record runs, you will need to set it as @liveonly. Please see tests for public access network setting. You're good to just resolve this comment.

File: src/azure-cli/requirements.py3.Darwin.txt [hunk 3]
```diff
@@ -64,7 +64,7 @@ azure-mgmt-netapp==8.0.0
 azure-mgmt-network==20.0.0
 azure-mgmt-policyinsights==1.1.0b2
 azure-mgmt-privatedns==1.0.0
-azure-mgmt-rdbms==10.0.0
+azure-mgmt-rdbms==10.1.0
```
Comment (DaeunYim) [inline src/azure-cli/requirements.py3.Darwin.txt:67]: Considering the 10.2.0b1 is already released, double check whether the new SDK has the byok parameters included, and make the SDK version change either in this PR or in server log PR. No need to update it twice.

Comment (alanenriqueo) [reply src/azure-cli/requirements.py3.Darwin.txt]: Checked with version 10.2.0b1 and everything worked OK.

File: src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_mysql.py (hunk 1 above)
Comment (alanenriqueo) [reply src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_mysql.py:358]: I have a check that makes the command fail if the server we are creating or updating has geo-redundant-backup enabled and data encryption.

Comment (alanenriqueo) [reply src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_mysql.py:359]: Now we are allowing data encryption with geo-redundant servers.

File: src/azure-cli/azure/cli/command_modules/rdbms/tests/latest/test_rdbms_flexible_commands.py [hunk 4]
```diff
@@ -507,6 +513,234 @@ def _test_flexible_server_georestore_mgmt(self, database_engine, resource_group)
...
+        replica_2_name = self.create_random_name(SERVER_NAME_PREFIX, SERVER_NAME_MAX_LENGTH)
+        backup_name = self.create_random_name(SERVER_NAME_PREFIX, SERVER_NAME_MAX_LENGTH)
+        tier = 'GeneralPurpose'
+        sku_name = 'Standard_D2ds_v4'
+        location = 'northeurope'
+        backup_location = 'westeurope'
+        backup_location_full_name = 'West Europe'
+
+        self.cmd(
+            'keyvault create -g {} -n {} --location {} --enable-soft-delete true --enable-purge-protection true'
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/rdbms/tests/latest/test_rdbms_flexible_commands.py:534]: Can you use `@KeyvaultPreparer(name_prefix='rdbmsvault', key='vault_name', location='northeurope', additional_params='--enabled-for-deployment true --enabled-for-template-deployment true')` instead?

**Review by evelyn-ys — APPROVED** [review]

**Pull request merged**